### DISCUSSION TITLE
Tariff commanded switching

### DISCUSSION THREAD

**Discussion by pkali1** [arendst/Tasmota#18080] [Ideas]
Dear all, I have ESP32 with three PZEM-016 installed in my DIY solar installation. I would like to send the command - switch Tariff1 or Tariff2 (and not set only timers for this). (one would be used for solar kWh and one for grid kWh counting). Is it somehow possible with this configuration?
Thanks.   Petr

**Comment by sfromis** [comment]
Not clear what you'd expect a command like "switch Tariff1" to do, but does not look like a current feature of Tasmota.

**Reply by pkali1** [reply]
Yes. I would like to switch Tariff by some command and not only by time, because I would like to measure into different counter when inverter is ON and OFF. So it would count Solar and Grid kWh. 

"switch Tariff1" or "switch Tariff2" would be perfect..  Or "Tariff toggle"...

**Reply by barbudor** [reply]
Such command doesn't exist now
It may be possible that dynamicaly changing times like sending a command to change the start time to now+1min

But I understand your need
So may be a variant such as 'Tariff1 ON' instead of a time to force Tariff1 could be a good solution (and the same for Tariff2 ON). 
I can have a look in the coming days unless @arendst prefer to do it himself.

**Reply by pkali1** [reply]
Thanks. Dynamic changing as workarround could be possible, but complicated a little.

**Reply by barbudor** [reply]
Improvement on going. I'll keep you posted

EDIT: PR https://github.com/arendst/Tasmota/pull/18102
Hopefully it will get merged as-is, or modified by Theo

**Reply by barbudor** [reply]
@pkali1 
PR has been merged
You should have Tasmota development binaries available with the new feature within 2 hours max.
Now working on the doc...

**Reply by pkali1** [reply]
Installed development into device. Starting test. Thanks a lot.

**Reply by pkali1** [reply]
Thanke you once again, working as expected. Very quick implementation. Now I can see on victron vrm both consumptions. Nice. Regards
